opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp

```diff
@@ -107,7 +109,10 @@ namespace Opm
         const Events& getEvents() const;
         const Deck& getModifierDeck(size_t timeStep) const;
         bool hasOilVaporizationProperties() const;
-
+        const VFPProdTable& getVFPProdTable(int table_id, size_t timeStep) const;
+        const VFPInjTable& getVFPInjTable(int table_id, size_t timeStep) const;
+        std::map<int, const VFPProdTable&> getVFPProdTables(size_t timeStep) const;
+        std::map<int, const VFPInjTable&>  getVFPInjTables(size_t timeStep) const;
```

Comment:
Is this supposed to be
```C++
map<int, const VFPInjTable>&
```
instead?  (Reference to map rather than map of references)

Reply:
Ehhh no - the map is a temporary structure, linking to tables elsewhere with application lifetime.

Reply:
> Ehhh no - the map is a temporary structure, linking to tables elsewhere with application lifetime.

That's a bug then.  You can't put bare references into a standard library container.  At **minimum** you need to make the elements into `std::reference_wrapper<T>`s.

Reply:
I have not figured out the problem, it looks like the table data got lost on the simulator side and what you mentioned is very likely to be the reason.

Reply:
> That's a bug then. You can't put bare references into a standard library container. 

If you say so; it compiles and runs tests without warning though?

> into std::reference_wrapper<T>s.

Or just use a pointer; I'll se.

Reply:
OK - I have updated the method to return a map of pointers; not married to that solution in any way.